kernel/e1000/src/lib.rs

```diff
@@ -458,17 +352,17 @@ impl E1000Nic {
         // because the `rx_cur` counter won't be able to catch up with the head index properly. 
         // Thus, we set it to one less than that in order to prevent such bugs. 
         // This doesn't prevent all of the rx buffers from being used, they will still all be used fully.
-        regs.rx_regs.rdt.write((E1000_NUM_RX_DESC - 1) as u32); 
+        rx_regs.update_rdt((E1000_NUM_RX_DESC - 1) as u32); 
         // TODO: document these various e1000 flags and why we're setting them
         regs.rctl.write(regs::RCTL_EN| regs::RCTL_SBP | regs::RCTL_LBM_NONE | regs::RTCL_RDMTS_HALF | regs::RCTL_BAM | regs::RCTL_SECRC  | regs::RCTL_BSIZE_2048);
 
         Ok((rx_descs, rx_bufs_in_use))
     }           
     
     /// Initialize the array of tramsmit descriptors and return them.
-    fn tx_init(regs: &mut E1000Registers) -> Result<BoxRefMut<MappedPages, [LegacyTxDescriptor]>, &'static str> {
+    fn tx_init(regs: &mut E1000Registers, tx_regs: &mut E1000TxQueueRegisters) -> Result<BoxRefMut<MappedPages, [LegacyTxDescriptor]>, &'static str> {
```

Comment:
this also needs reformatting. Use this format for any "long" functions, lets  say beyond 100 chars wide:
```rust
fn tx_init(
    regs: &mut E1000Registers,
    tx_regs: &mut E1000TxQueueRegisters
) -> Result<BoxRefMut<MappedPages, [LegacyTxDescriptor]>, &'static str> {
    ....
}
```

Reply:
one day i'll set up rustfmt... lol